faros_event.sh

```diff
@@ -402,432 +345,280 @@ function resolveDefaults() {
+                "environment": $deploy_env,
+                "id": $deploy_id
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_status" ]; then
+        request_body=$(jq \
+            --arg deploy_status "$deploy_status" \
+            '.data.deploy +=
+            {
+                "status": $deploy_status
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_status_details" ]; then
+        request_body=$(jq \
+            --arg deploy_status_details "$deploy_status_details" \
+            '.data.deploy +=
+            {
+                "statusDetails": $deploy_status_details
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_env_details" ]; then
+        request_body=$(jq \
+            --arg deploy_env_details "$deploy_env_details" \
+            '.data.deploy +=
+            {
+                "environmentDetails": $deploy_env_details
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_start_time" ]; then
+        request_body=$(jq \
+            --arg deploy_start_time "$deploy_start_time" \
+            '.data.deploy +=
+            {
+                startTime: $deploy_start_time|tonumber
+            }' <<< $request_body
+        )
+    fi
+    if ! [ -z "$deploy_end_time" ]; then
+        request_body=$(jq \
+            --arg deploy_end_time "$deploy_end_time" \
+            '.data.deploy +=
+            {
+                endTime: $deploy_end_time|tonumber
+            }' <<< $request_body
+        )
+    fi
 }
 
-function addCDObjectsToEvent() {
-    makeDeployment
-    makeArtifactDeployment
-    makeApplication
-    request_body=$(jq ".entries += [
-        $cicd_Deployment,
-        $cicd_ArtifactDeployment,
-        $compute_Application
-    ]" <<< $request_body)
+function addCommitToData() {
+    if ! [ -z "$commit_sha" ] && 
+       ! [ -z "$commit_repo" ] &&
+       ! [ -z "$commit_org" ] && 
+       ! [ -z "$commit_source" ]; then
+        request_body=$(jq \
+            --arg commit_sha "$commit_sha" \
+            --arg commit_repo "$commit_repo" \
+            --arg commit_org "$commit_org" \
+            --arg commit_source "$commit_source" \
+            '.data.commit +=
+            {
+                "sha": $commit_sha,
+                "repository": $commit_repo,
+                "organization": $commit_org,
+                "source": $commit_source
+            }' <<< $request_body
+        )
+    fi
 }
 
-function addCIObjectsToEvent() {
-    makeArtifact
-    makeArtifactCommitAssociation
-    request_body=$(jq ".entries += [
-        $cicd_Artifact,
-        $cicd_ArtifactCommitAssociation
-    ]" <<< $request_body)
-}
+function addArtifactToData() {
+    if ! [ -z "$artifact_source" ] &&
+       ! [ -z "$artifact_org" ] &&
+       ! [ -z "$artifact_repo" ] &&
+       ! [ -z "$artifact_id" ]; then
+        request_body=$(jq \
+            --arg artifact_source "$artifact_source" \
```

Comment:
**nit:**
you could use shorter names on the jq args (applied evywhere) e.g
```
--arg source "$artifact_source" \
```
and then
```
"source": $source,
```